docs/developers/doxygen_style_guide.rst

```diff
@@ -0,0 +1,122 @@
+..
+  SPDX-License-Identifier: CC-BY-4.0
+  Copyright Contributors to the OpenColorIO Project.
+
+.. _doxygen_style_guide:
+
+OpenColorIO Doxygen Style Guide
+===============================
+
+This guide introduces a consistent style for documenting OCIO public C++ headers
+with Doxygen. New code should be documented with the following conventions.
```

Comment:
We used Doxygen in the SynColor library that was the precursor to our work on OCIO.  Personally I found that it made the header files very verbose and harder to read.  Following these rules strictly results in a lot of lines that arguably have no value.  E.g.  

\brief Sets the name of the foo.
\param fooName The new name for the foo.
void setFooName(char * fooName);

... and so forth.  Is this really useful?  Clearly this will be a big change for the original OCIO headers since these obvious functions are totally undocumented.  I'm not opposed to being religious about it if that's what people want, but I think we should have some discussion about what to do for the obvious cases.  The style guide should then provide guidance about how to handle this and how strictly the rules are expected to be followed in cases where the function/method and variable names themselves already convey the necessary information.

Reply:
I kind of prefer being very verbose with documentation, otherwise it becomes subjective about what should be documentation and what shouldn't i.e. what is "obvious". Getter/Setters are probably the exception as you said, so sure we can come up with some rules about that.